### DISCUSSION TITLE
no CTRL menu within Win10 as an „exe“, but under macOS okay!?!

### DISCUSSION THREAD

**Discussion by hps2109** [RomanKubiak/ctrlr#659] [General]
Hello to you all!
The panel works perfectly under macOS, but not under Win10.

The same *.bpanelz, (created with macOS) was loaded and export as „Export as restricted instance“.
I tried everything, I could find in the discussions, but it just doesn't work :-(
There's definitely an undocumented trick, right?
Needed is the midi input „Device & Channel“ & midi output „Device & Channel“ functions.
Otherwise the panel makes no sense under Win.

Btw… is it basically possible to switch off, one or the other menu function?

Thank you so much for your help!

Kind regards
Peter
![macOS Kurzweil 1000 Series (Remote)](https://github.com/RomanKubiak/ctrlr/assets/167638645/664b39ad-462f-4699-8eff-d768ffab3de0)
![Win10_no menu](https://github.com/RomanKubiak/ctrlr/assets/167638645/2b9f1b2b-2ad1-407b-93ba-a626b9ca694d)
[Kurzweil 1000 XX Series_1_3_2024-06-17_12-13.bpanelz.zip](https://github.com/user-attachments/files/16019672/Kurzweil.1000.XX.Series_1_3_2024-06-17_12-13.bpanelz.zip)

**Comment by damiensellier** [comment]
I guess it's an old version of Ctrlr, they have a problem hiding the bar (height 0px) when exporting an instance.

I suggest you use the latest version of CtrlrX, this menuBar problem is fixed.

https://github.com/RomanKubiak/ctrlr/discussions/619#discussioncomment-8779822

**Reply by damiensellier** [reply]
You also have options on what is going to be shown when exporting an instance in the global panel property pane:  

![Capture d’écran, le 2024-06-27 à 15 42 34](https://github.com/RomanKubiak/ctrlr/assets/55164385/fea7655b-ca28-4f4e-a25b-06a21c005e45)

**Reply by hps2109** [reply]
Hi Damien,
thank you for your information!
It was my mistake to inform, that version V5.6.30.1 is installed on my iMac (Intel) and version V5.6.30.0 on the Win10 PC (Intel) from a friend of mine.
Sure, I will give it a "go".
Many thanks for your info!

Kind regards
Peter

**Reply by damiensellier** [reply]
Do you mean that the menu bar was not showing up on the Win10 exported instance with 5.6.30?

**Reply by hps2109** [reply]
Unfortunately, yes. The current version is installed on both PC's and checked the options.
I also exported (restricted instance) a ctrl demo panel as a test.
Nothing happens. The exported (restricted instance) *.exe App doesn't show the menu bar.

**Reply by damiensellier** [reply]
I'll investigate why it's working on mac and not on pc. I only use mac for developement and usually hide menu bars on exported panels.
I have an idea where it could came from so stay tuned.

**Reply by damiensellier** [reply]
Could you please check if it's working with older versions for Windows like 5.3.201?

**Reply by hps2109** [reply]
Great!
Yes sure, I will install the version 5.3.201 additionally and tested it.

**Reply by damiensellier** [reply]
There used to be this trick where you added 24px to the canvas size to show the hidden menuBar. Maybe you can try that too. Iet me check so that I post the link to this trick.

**Reply by damiensellier** [reply]
just add 24 pixels to the canvas bounds height (last of the four numbers in one) before exporting with older versions.
for exemple if the panel is 600px high, set the canvas bounds to 0 0 800 624 before exporting. (this won't work with version 5.6.23+)

you can also force the bar to show up with a lua script launched when the panel is ready: 

`panel:getPanelEditor():setProperty("uiPanelMenuBarVisible", 1, false)`

https://ctrlr.org/forums/topic/export-restricted-instance-hidden-menu-bar/
https://ctrlr.org/forums/topic/hidingshowing-menu-bar/

**Reply by damiensellier** [reply]
Worst case scenario, if you just need access to the midi settings you can assign it to a button on value change : 
<img width="558" alt="Capture d’écran 2024-06-28 à 06 06 45" src="https://github.com/RomanKubiak/ctrlr/assets/55164385/a5eba75b-299e-4461-b0ab-102952a2c875">

https://github.com/RomanKubiak/ctrlr/discussions/365

**Reply by hps2109** [reply]
Hi Damien,
your suggestion to install version V5.3.201 was successful.
![Screenshot (CTRL 5 3 201)](https://github.com/RomanKubiak/ctrlr/assets/167638645/7969a8eb-f74a-4955-8a2a-dc3243a4f556)

The menu bar is now visible! wow! But I forgot to add the 24pixels,  will check it now too. :-)

**Reply by hps2109** [reply]
added 24 pixels :-)
![Screenshot (CTRL 5 3 201 + 24 pixel)](https://github.com/RomanKubiak/ctrlr/assets/167638645/fa849482-95ce-405b-88ec-3646a9324273)

Thanks a lot for your additional tips!

**Comment by dobo365** [comment]
Hi Peter,
  I'm working with Damien on CtrlrX.

It works fine to get the menu on PC or Mac version with Ctrlr 5.6.30.
I guess you forgot to switch OFF the panel property "Hide the menu bar when exporting an instance".
I had the case myself when doing an export with a new version and thought it was the coding as my panel was the same.
Then I remembered about that property and saw it had been switched ON.
Will discuss and check that with Damien. Should be OFF by default I think (even if I will soon export all my panels without menu).

**Comment by hps2109** [comment]
Hi goodweather,
many thanks for your reply! Yes, I know. Read a lot from you and Damien in the discussions.

Just checked it again with CTRL-5-6-30-1.
The checkbox is unselected and if I export it to an „Export as restricted instance“,
The menu bar isn’t visible in the program.
I use the same *.bpanelz file that I created under macOS (10.13).
At first, I thought the files were not compatible, but why should that be?

Maybe I should hide the entire menu too, all I need is the MIDI configuration and perhaps zoom in/out.
I'm still thinking about how I can do this with a pop window

Cheers
Peter

![Hide the menu bar - off](https://github.com/RomanKubiak/ctrlr/assets/167638645/1de3a864-3d1f-4029-8dc8-846c4dd10559)

**Comment by damiensellier** [comment]
@hps2109 is absolutely right.
I just checked on my PC with ctrlrx 5.6.30.1 exe available on the release discussion page and it's been built with wrong files. Not sure if it's the JUCE addons missing or the .xml missing but the LnF are not working properly and the menuBar is not showing up on export like it's supposed to do.

@dobo365 Could you please post your exported build here so than I can fix the links to this version everywhere. I'd need VST VST3 and EXE for intel PC

SO

Before we get the working v5.6.30.1 version working, you can use 5.6.29. it exports the instances with the menuBar as it should.

Windows 10
[CtrlrX-x64-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216062/CtrlrX-x64-5.6.29.zip)
[CtrlrX-x64.vst3-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216131/CtrlrX-x64.vst3-5.6.29.zip)

**Reply by dobo365** [reply]
OK. Only got some time now...
All my last versions were also not working. Finally I deleted Ctrlr.settings and things are back...

So, I made some tests with the different versions I downloaded
5.6.29 and 5.6.30 are OK. 
5.6.30.1 doesn't display the menu and shifts the panel 24px higher leaving a blank area at the bottom
My own 5.6.30.1F works fine as well.
I will do the change to get the menu in Midi Monitor and Log viewer to produce a 5.6.30.1G
I didn't download the source for 5.6.29, 5.6.30. Only 5.6.30.1 that I modified. Is it OK to use that for your Standalone, VST and VST3?

**Reply by damiensellier** [reply]
Hi Dominique, Sure let's go for the rev.1G :) 
Thanks in advance, I'll update the links above asap.

EDIT :  v5.6.30.3 would be better...
